docs/docs/dynamo-store/access-strategies.md

```diff
@@ -0,0 +1,73 @@
+# Access Strategies
+
+The access strategy allows you to opt into custom loading or reloading
+algorithms. For almost all categories in almost all most systems, `Unoptimized`
+is a good choice; you can always switch later.
+
+
+# Unoptimized
+
+The first, and likely most used access strategy is the unoptimized one. It'll
+not apply any optimised reading logic. When loading a stream for the first time
+all of its events will be loaded in batches of `batchSize` (default: `500`).
+This turns out to be a sensible default for most aggregates, especially when
+used in conjunction with the built-in caching layer.
+
+# LatestKnownEvent
+
+There is a special case of stream whose events are all a "full-replace". An
+example could be a customer's contact preferences, or an entity view-data
+stream. This access strategy will ensure that you only load at most a single
+event when transacting or querying such a stream.
+
+# Snapshot(isOrigin, toSnapshot)
+
+Imagine a simple counter service. It does not have a defined start and end as
+most processes would and as such we don't know how large it'll grow over time.
+In order to ensure the snappiness of our application we might apply
+snapshotting to ensure that the state of the counter can be reconstructed in 2
+round-trips instead of N, where N is the number of events divided by batch
+size.
+
+In Equinox the snapshot is a member of the Event DU
+
+```ts
+type Event =
+  | { type: "Increment", data: { amount: number } }
+  | { type: "Snapshot", data: { current: number } }
+
+const codec = Codec.json<Event>()
+```
+
+We'll handle the snapshot in our evolution function
+
+```ts
+type State = number
+const initial = 0
+const evolve = (state: State, event: Event) => {
+  switch (event.type) {
+    case "Increment": return state + event.data.amount
+    case "Snapshot": return event.data.current
+  }
+}
+const fold = reduce(evolve)
+```
+
+In addition to this we need to specify to Equinox which event type represents a
+snapshot and how to transform the current state into a snapshot
+
+```ts
+const isOrigin = (ev: Event) => ev.type === "Snapshot"
+const toSnapshot = (state: State): Event => 
+  ({ type: snapshotEventType, data: { current: state } })
+```
+
+We can then create the access strategy
+```ts
+const access = AccessStrategy.Snapshot(isOrigin, toSnapshot)
+```
+
+With this in place Equinox will maintain and store a snapshot event in the tip
+document that is guaranteed to always be up to date, and as a result the state
```

Comment:
item is prob better than document in DDB terms?
or maybe mention that there are conceptually pages/chapters, and you're looking in the conclusion chapter all the time

Reply:
Good point, my terminology is all over the place. Document/Item/Tip/Stream/... Should pick one and go with it

Maybe I should read https://www.fernandomc.com/posts/dynamodb-key-terms-explained/

Reply:
should standardise on partition and sort per https://stackoverflow.com/questions/12241235/dynamodb-query-versus-getitem-for-single-item-retrieval-based-on-the-index#comment131304321_12248430
and if you're bored you can make sure the FS versions does that too!